app/services/dttp/retrieve_trn.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Dttp
+  class RetrieveTrn
+    include ServicePattern
+
+    class HttpError < StandardError; end
+    class TraineeAttributeError < StandardError; end
+
+    POLL_INTERVAL = 6.hours
+    MAX_POLL_DURATION = 2.days
+
+    attr_reader :trainee
+
+    def initialize(trainee:)
+      @trainee = trainee
+    end
+
+    def call
+      trn = fetch_trn
+
+      return trainee.update!(trn: trn) if trn
+
+      RetrieveTrnJob.set(wait: POLL_INTERVAL).perform_later(trainee.id) if continue_polling?
```

Comment:
I think we're mixing concerns here. If we do it like this the service has to be aware that it is being called by a job. I think the service should just return either the TRN or nil and the calling job can then decide to update the trainee or queue another job. This service wouldn't need to know about the trainee particularly either could just take a dttp_id argument and return a TRN or not.